File: src/Easy/Leetcode226.java
package Easy;

import java.util.LinkedList;
import java.util.Queue;

//BFS
//Maybe referred approach 2
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

public class Leetcode226 {
    public TreeNode invertTree(TreeNode root) {
        Queue<TreeNode> queue=new LinkedList<>();
        if(root==null){
            return null;
        }
        queue.add(root);
        while(!queue.isEmpty()){
            TreeNode front=queue.poll();
            TreeNode temp=front.left;
            front.left=front.right;
            front.right=temp;
            if(front.left!=null){
                queue.add(front.left);
            }
            if(front.right!=null){
                queue.add(front.right);
            }
        }
        return root;
    }
}

File: src/Python/Easy/Leetcode226.py
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root is None:
            return None

        bfs_queue=[]
        bfs_queue.append(root)
        while bfs_queue:
            front=bfs_queue.pop()
            temp=front.left
            front.left=front.right
            front.right=temp
            if front.left is not None:
                bfs_queue.append(front.left)
            if front.right is not None:
                bfs_queue.append(front.right)

        return root